spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
Yes, that makes sense too.
Let's leave as is and I'll polish that minor Doc issue on merge!
After all I even don't care about that possible second session instance here!